packages/rich-text-editor/src/RichTextEditor/controlmap.ts

```diff
@@ -37,36 +38,67 @@ function toggleMarkCommand(mark: MarkType): Command {
   ) => toggleMark(mark)(state, dispatch)
 }
 
+// TODO: If a different list is clicked on a list update the list type
 function toggleListCommand(node: NodeType): Command {
   return (
     state: EditorState,
     dispatch: ((tr: Transaction) => void) | undefined
   ) => wrapInList(node)(state, dispatch)
 }
 
-function liftOrIndentList(action: "lift" | "indent"): Command {
+function liftList(): Command {
   return (
     state: EditorState,
     dispatch: ((tr: Transaction) => void) | undefined
   ) => {
     const { $from } = state.selection
     // calculate the parent node from the current tag selected
-    const topLevelListNode = $from.node($from.depth - 1)?.type
-    const islistNode =
-      (topLevelListNode?.name && topLevelListNode.name === "listItem") || false
+    const listItemNode = $from.node($from.depth - 1)?.type
 
-    if (!islistNode) {
-      return false
-    }
+    return liftListItem(listItemNode)(state, dispatch)
+  }
+}
 
-    if (action === "lift") {
-      return liftListItem(topLevelListNode)(state, dispatch)
-    } else {
-      return sinkListItem(topLevelListNode)(state, dispatch)
-    }
+// increase list indent should only be available on the second list node of a list item (otherwise it should be disabled)
+function indentList(): Command {
+  return (
+    state: EditorState,
+    dispatch: ((tr: Transaction) => void) | undefined
+  ) => {
+    const { $from } = state.selection
+    const listItemNode = $from.node($from.depth - 1)?.type
+
+    return sinkListItem(listItemNode)(state, dispatch)
   }
 }
 
+// If there is a valid list item its indent can be decrease or 'lifted'
+function liftListIsDisabled(state: EditorState): boolean {
+  const { $from } = state.selection
+  const listItemNode = $from.node($from.depth - 1)?.type
+  const isValidListItem = listItemNode?.name === "listItem" || false
+
+  return !isValidListItem
+}
+
+// If the is a valid list item and it is not the first in a list it can be indented
```

Comment:
`If there*` meant to be here I think?

Reply:
or is it thee? (yes it is meant to be `there `😭 )